resources/qml/delegates/ImageMessage.qml

```diff
@@ -31,11 +31,40 @@ Item {
         fillMode: Image.PreserveAspectFit
 
         MouseArea {
+            id: mouseArea
             enabled: model.data.type == MtxEvent.ImageMessage && img.status == Image.Ready
+            hoverEnabled: true
             anchors.fill: parent
             onClicked: TimelineManager.openImageOverlay(model.data.url, model.data.id)
         }
 
-    }
+        Item {
+            id: overlay
+        
+            anchors.fill: parent
+            visible: mouseArea.containsMouse
+            
+            Rectangle {
+                id: container
+                width: parent.width
+                implicitHeight: imgcaption.implicitHeight
+                anchors.bottom: overlay.bottom 
+                color: "black"
+                opacity: 0.75
+
+                Text {
+                    id: imgcaption
 
+                    anchors.fill: parent
+                    elide: Text.ElideMiddle
+                    horizontalAlignment: Text.AlignHCenter
+                    verticalAlignment: Text.AlignVCenter
```

Comment:
I think that one is unnecessary, since this should only be the height of the text anyway?

Reply:
Tested out each line and the anchors.fill ensures the text fills the `container` rectangle, and the two alignment properties center the text which I felt was the most aesthetic

Reply:
I don't think you need the vertical alignment, since the rectangle should only be as high as the text, but I don't think it actually makes a difference.